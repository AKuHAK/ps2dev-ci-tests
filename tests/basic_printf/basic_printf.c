#include <stdio.h>

#include <kernel.h>
#include <ps2_poweroff_driver.h>
#include <sbv_patches.h>
#include <libpwroff.h>
#include <sifrpc.h>

int main(void)
{
	SifInitRpc(0);
	sbv_patch_enable_lmb();
	sbv_patch_disable_prefix_check();
	init_poweroff_driver();

	printf("===!BEGIN\n");

	printf("Hello, World!\n");
	printf("%d\n", 42);
	printf("%s\n", "Hello, World!");
	printf("%X\n", 0xdeadbeef);
	printf("%x\n", 0xdeadbeef);

	printf("===!END\n");


	poweroffShutdown();
	return 0;
}
