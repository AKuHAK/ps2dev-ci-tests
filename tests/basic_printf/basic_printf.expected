Hello, World!
42
Hello, World!
DEADBEEF
deadbeef
